### ISSUE TITLE
Damage required to turn a body part red is too high in advanced medical

### ISSUE THREAD

**Issue by alganthe** [acemod/ACE3#4347]
**Arma 3 Version:** `1.64` RC
**CBA Version:** `3.0.0`
**ACE3 Version:** `3.7.0` release; commit: f3dc1afdbfc8dc05c6057a5d6806c1a7cc10e6db

**Mods:**
- `@CBA_A3`
- `@ace`

**Description:**
- The amount of damage required to turn a body part icon (the crosses) red is too high, 2 large wounds are still shown as yellow when before a single large wound was red.
- http://puu.sh/r1mTx/671444d92a.jpg

**Comment by PabstMirror** [comment]
https://github.com/acemod/ACE3/blob/master/addons/medical/functions/fnc_modifyMedicalAction.sqf#L31

Right now the yellow/red switch is at bleeding rate of 0.15.  
Below 60 will cause unit to passout, so that is about 4.5 minutes.

**Comment by jonpas** [comment]
So that's entirely blood related, even I didn't know that. I guess this can be closed then?

**Comment by PabstMirror** [comment]
I think you could argue that the value could be lower
I just picked 0.15 because it seemed reasonable, but 0.1 might be good as well

2 large Avulsions will add to 0.1

**Comment by alganthe** [comment]
actually the body representation is accurate, it's just the interaction icons that don't have the correct color:
http://puu.sh/r1NTB/6ad1afcc13.jpg

**Comment by crowmium** [comment]
Maybe add another intermediate color(s)? Like orange? Usually wounds are described on a scale and not just minor or major. The AIS, an injury scale for traumatic injuries, uses 6 levels for instance: https://en.wikipedia.org/wiki/Abbreviated_Injury_Scale#Severity.

Maybe you could assign various colors like: Yellow-green, Yellow, Orange, Red, and Black for Minor, Moderate, Serious, Severe, and Critical wounds respectively?

**Comment by ace3mod** [comment]
Hello @acemod/maintainers. This ticket has been open for over 163 days without any activity.

**Comment by SilentSpike** [comment]
Looking into this now, it's a pity the action menu doesn't support colours as we could then just use the same code for both.

Here's the body image colour code:

```sqf
private _red = 1;
    private _green = 1;
    private _blue = 1;

    private _torniquet = _selectionTourniquet select _forEachIndex;
    if (_torniquet > 0) then {
        _red = 0;
        _green = 0;
        _blue = 0.8;
    } else {
        private _bloodLoss = _selectionBloodLoss select _forEachIndex;
        if (_bloodLoss > 0) then {
            _green = 0 max (0.8 - _bloodLoss);
            _blue = _green;
        } else {
            private _damage = _selectionDamage select _forEachIndex;
            if (_damage > 0.1) then {
                _blue = 0 max (1 - _damage);
                _green = 0.5 max (1 - _damage / 2);
            };
        };
    };
```

**Comment by SilentSpike** [comment]
Something doesn't add up, because if the bloodloss is 0.15 the body image will be approximately #FFA6A6 (a very pale red) so it should be a fine threshold for the icon to change colour 🤔

**Comment by SilentSpike** [comment]
Found the discrepancy.

How bloodloss is calculated to be fed into the body image function:
```sqf
    _selectionBloodLoss set [_bodyPartN, (_selectionBloodLoss select _bodyPartN) + (20 * _bleeding * _amountOf)];
```

Whereas the interaction icon just does:
```sqf
_bloodLossOnBodyPart = _bloodLossOnBodyPart + (_amountOf * _bleeding);
```

However, I don't really get how the former works because the 20 seems to make the values too big to work with the colour calculation code above 😕

**Comment by SilentSpike** [comment]
Ah wait, I get it now, it's because the `_bleeding` and `_amountOf` percentages compound together, so the 20 scales it back up. Seems like an arbitrary value though, but simply adding this to the icon code should make the colour change correctly.

**Comment by SilentSpike** [comment]
Fixed in #6831
